hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java

```diff
@@ -620,8 +622,22 @@ private CheckAndMutateResult checkAndMutate(HRegion region, List<ClientProtos.Ac
             spaceQuotaEnforcement.getPolicyEnforcement(region).check(del);
             mutations.add(del);
             break;
+          case INCREMENT:
+            Increment increment = ProtobufUtil.toIncrement(action.getMutation(), cellScanner);
+            ++countOfCompleteMutation;
+            checkCellSizeLimit(region, increment);
+            spaceQuotaEnforcement.getPolicyEnforcement(region).check(increment);
+            mutations.add(increment);
+            break;
+          case APPEND:
+            Append append = ProtobufUtil.toAppend(action.getMutation(), cellScanner);
+            ++countOfCompleteMutation;
+            checkCellSizeLimit(region, append);
+            spaceQuotaEnforcement.getPolicyEnforcement(region).check(append);
+            mutations.add(append);
+            break;
           default:
-            throw new DoNotRetryIOException("Atomic put and/or delete only, not " + type.name());
+            throw new AssertionError("invalid mutation type : " + type);
```

Comment:
Why change to AssertionError?

Reply:
I think what you said is also reasonable, but since this is an rpc method, we can not make sure whether the remote side has the same code with us at server side, so I still prefer a DoNotRetryIOException here. Let's not change it for now.

Reply:
Sure. I will revert this change.